pkg/storage/unified/sql/backend.go

```diff
@@ -650,15 +650,23 @@ func (b *backend) poll(ctx context.Context, grp string, res string, since int64,
 // in a single roundtrip. This would reduce the latency of the operation, and also increase the
 // throughput of the system. This is a good candidate for a future optimization.
 func resourceVersionAtomicInc(ctx context.Context, x db.ContextExecer, d sqltemplate.Dialect, key *resource.ResourceKey) (newVersion int64, err error) {
-	// TODO: refactor this code to run in a multi-statement transaction in order to minimize the number of round trips.
-	// 1 Lock the row for update
-	rv, err := dbutil.QueryRow(ctx, x, sqlResourceVersionGet, sqlResourceVersionRequest{
+
+	// 1. Increase the resource version to the current epoch with microseconds precision.
+	// This is an update statement that will lock to row and prevent concurrent updates until the transaction is commited.
+	//
+	// Note: there is a small chance that the resource version will be the same as the previous one when
+	// concurrent updates are too close in time. This is due to the fact that the current epoch is unlikely to cause
+	// real issues in practice (given network latency, etc).
+	// In rare occasions (like master clock changes) the resource version might go backwards for a short period of time.
```

Comment:
> like master clock changes
1. Is the reason this happens to adjust for clock drift?
2. How often does this happen? 
3. And when it does, how much time are we talking about?

Reply:
Removed this comment based on Dan's suggestion to ensure the counter never goes back in time.

To answer:
1/ Yes ... ntp clock synchronisation happens all the time on servers to ensure clock doesn't drift too much.
2/ It's adjusted in the background by an ntp server. Generally every few minutes at most
3/ When time is adjusted, time might go back/ahead by a few milliseconds, but can be  tens or hundreds of ms from time to time.